### ISSUE TITLE
"Link Cels" could link the first filled Cel in a selection, instead of the first Cel in-sequence, maybe?

### ISSUE THREAD

**Issue by oceanhahn** [aseprite/aseprite#2137]
This is a suggestion I am less sure about but would consider pretty okay:

Currently, "Link Cels" duplicates the _first_ Cel in a range over every other selected Cel in that range, if the lowest-numbered Frame's Cel has content. If it is vacant, the function whiffs. What I propose is to have the first _non-vacant Cel_ in a selected sequence used as the source, rather than doing nothing. (For instance, if Cels in Frames 0~12 are selected, 0 and 1 are vacant, and 2 has content, the whole range fills with linked-duplicates of Frame 2's Cel.)

Additionally, this behaviour would also help when attempting to link previous Cels (per #1361).

My only concern is that it could be outside of what come users might expect when using Link Cels to _overwrite downsequence Cels,_ when the first nonvacant Cel is reproduced _instead of_ the Cel in the active Frame (which, although not current behaviour, is also something I would consider an improvement).

Anyway, I know this is a mess, but thank you for reading all the way to the end, and for giving it your consideration! :D

..using ASEprite 1.2.13-x86 on Win10.

**Comment by dacap** [comment]
After reading this I understood #1361 too 😅 I'll fix this for v1.2.14 👍

**Comment by oceanhahn** [comment]
I am pleased that it was so useful to you! :D

**Comment by oceanhahn** [comment]
Just to make sure we are considering every option here, it may be worth considering using the _first selected Cel_ as the template, rather than the lowest-numbered nonvacant one. This would probably be harder to implement, but I think it is worth considering from a user-predictable behaviour standpoint. Perhaps duplicating the Cel in the **current (visible) Frame** _or,_ if the current Frame is vacant, the lowest-numbered nonvacant Cel?

This is not really a preference as much as a consideration-- I am sure whatever it ends up being will be good! :D

**Comment by dacap** [comment]
Yeah, it's getting complex, first I thought the *lowest-numbered non-empty cel*, but now I guess that you are right, it should behave in a different way. Let's think 🤔

1. There is a little problem with the popup menu, when we right-click a specific cel, the active cel doesn't change, so there might be a confusion here: should we link the active cel or the right-clicked cel? (e.g. if both are non-empty)
2. Should we change the active cel with a right-click (I'm not sure about this one)
3. Other more intuitive possibility would be that we should be able to resize the range of cels from the left or right side, something like this:

![resize-cel](https://user-images.githubusercontent.com/39654/63649695-39d5d100-c717-11e9-9bc3-0abf6f12c8ef.gif)

**Comment by oceanhahn** [comment]
I like (3) but I worry that it might interfere with users' ability to _move_ individual (or ranges of) Cels. It also leaves to question what would happen if a selection of Cels is active when the user wants to try and 'stretch' the linked Cels over..

For (1), perhaps right-clicking switches the active Frame to the corresponding Cel, and links that? Is it possible to perhaps add a simple GUI element arrow pointing outward from the template Cel through the range (or towards the overwritten Cel, if there is no selection)? It might add a bit more clutter to the Timeline but it would leave no ambiguity about which Cel is the source.
![image](https://user-images.githubusercontent.com/11544843/63654186-f6925700-c744-11e9-9256-21201cdac127.png)

And for (2), I feel like switching the active _Frame_ is a good plan, without disrupting the selection in the Timeline. A problem I can foresee with this though is if a Frame _outside_ the selection is RMB-invoked, are the relevant Cel(s) from that Frame used as the template in the selection _even though they are not members of it_ (for consistency's sake, I would argue yes), and are the newly-linked duplicates _also linked to the Cels outside the selection_ (very weak 'no' from me on this side)?
